web/e2e/page-objects/browse/browse.ts

```diff
@@ -73,13 +119,84 @@ export class BrowsePage extends AppPage {
 
   async clickFacetName(collection: string) {
     let facet = this.facetName(collection);
-    return await facet.click();
+    await facet.click();
+    await this.waitForSpinnerDisappear();
   }
 
   facetCount(collection: string) {
     return element(by.cssContainingText(".facet-list .facet-value .facet-name-list", collection))
       .element(by.xpath("following-sibling::span"));
   }
+
+  async closeCollection() {
+    await element(by.css(".mdl-button > .fa-remove")).click();
+    await this.waitForSpinnerDisappear();
+  }
+
+  get searchingSpinner() {
+    return element(by.css(".ng-star-inserted > h3"));
+  }
+
+  //protractor wait method doesn't work well for spinner elements.
+  //Using for loop solves the issue.
+  async waitForSpinnerDisappear() {
+    for (let i = 0; i < 100; i++) {
+      if (this.searchingSpinner.isPresent()) {
+        await browser.wait(EC.not(EC.presenceOf(element(by.css(".ng-star-inserted > h3")))));
+        break;
+      }
+      await browser.sleep(100);
+    }
+  }
+
+  /**
+   *
+   * @param page: first, previous, next, last
+   */
+  async paginateClick(page: string) {
```

Comment:
for naming consistency with the rest, rename the function to: clickPaginate()

Reply:
Fixed.